### ISSUE TITLE
Cannot export edited Basic Revolution Animation (.brlan) files

### ISSUE THREAD

**Issue by SuperFromND** [KillzXGaming/Switch-Toolbox#526]
I know Wii stuff is a lowish priority for this tool given its name, but I thought it'd be worth writing an issue anyways in case someone else runs into this problem.

I'm trying to make use of Switch Toolbox's robust featureset and support of .brlan and .brlyt files to create some custom banner animations for Wii homebrew forwarding channels, as there doesn't appear to be any other good tools for this purpose. However, while Switch Toolbox is able to read the files and allows me to modify them, I am unable to save any of my edits in any way as the program crashes when trying to export my modified BRLAN file.

Here's the process I'm using to create the .brlan file I'm trying to edit:

- First, grab a Wii WAD (I'm using [this static base WAD from CustomizeMii](https://github.com/Brawl345/customizemii/blob/master/Base_WADs/StaticBase.wad); I'm to believe this is okay to share as it's a forwarder channel created specifically to modify and redistribute, and was made seemingly from scratch) and use [ShowMiiWads](https://wiibrew.org/wiki/ShowMiiWads) to extact its contents.
- Once it's done extracting, use the [U8Mii tool included in Wii.cs tools](https://code.google.com/archive/p/showmiiwads/downloads) to unpack `00000000.app` from that folder; this will create a new folder containing a few files. 
- In this case, use U8Mii again to unpack `meta/banner.bin`, then use U8Mii one more time to pack the output of `meta/banner.bin` into a new `.bin` file. (This is because of a minor but separate issue that Switch Toolkit cannot read an extracted Wii Channel's `banner.bin` file as a U8 archive, likely due to the IMET header at the start.)
- Once all of that is done, open this new `banner.bin` file in Switch Toolbox, then open either `arc/blyt/banner.brlyt` or `arc/blyt/banner.brlan` to open the Layout Editor.
- Select the Animation Heirarchy tab and right-click `banner.brlan`, then click "Export". Switch Toolbox should then crash.

And here's the relevant error information from the error dialog that appears when trying to export:
```
************** Exception Text **************
System.ArgumentNullException: Value cannot be null.
Parameter name: buffer
   at System.IO.BinaryWriter.Write(Byte[] buffer)
   at LayoutBXLYT.BRLAN.PAT1.Write(FileWriter writer, LayoutHeader header) in C:\projects\switch-toolbox\File_Format_Library\FileFormats\Layout\Rev\BRLAN.cs:line 255
   at LayoutBXLYT.LayoutHeader.WriteSection(FileWriter writer, String magic, SectionCommon section, Action WriteMethod) in C:\projects\switch-toolbox\File_Format_Library\FileFormats\Layout\Common.cs:line 1646
   at LayoutBXLYT.BRLAN.Header.Write(FileWriter writer) in C:\projects\switch-toolbox\File_Format_Library\FileFormats\Layout\Rev\BRLAN.cs:line 180
   at LayoutBXLYT.LayoutAnimList.exportToolStripMenuItem_Click(Object sender, EventArgs e) in C:\projects\switch-toolbox\File_Format_Library\GUI\BFLYT\Animations\LayoutAnimList.cs:line 152
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```
